server/plugin/api.go

```diff
@@ -245,6 +252,168 @@ func (p *Plugin) handleUserDisconnect(w http.ResponseWriter, r *http.Request) {
+	submitRequest := &model.SubmitDialogRequest{}
+	if err := decoder.Decode(&submitRequest); err != nil {
+		p.API.LogError("Error decoding SubmitDialogRequest.", "Error", err.Error())
+		p.returnSubmitDialogResponse(w, response)
+		return
+	}
+
+	ctx := r.Context()
+	token := ctx.Value(ContextTokenKey).(*oauth2.Token)
+	userID := r.Header.Get(HeaderServiceNowUserID)
+	var selectedOption string
+
+	if len(strings.Split(submitRequest.CallbackId, "__")) != 2 {
+		p.API.LogError(InvalidCallbackIDError)
+		response.Error = InvalidCallbackIDError
+		p.returnSubmitDialogResponse(w, response)
+		return
+	}
+
+	postID := strings.Split(submitRequest.CallbackId, "__")[0]
+	inputType := strings.Split(submitRequest.CallbackId, "__")[1]
+
+	switch inputType {
+	case DateTimeUIType:
+		selectedOption = fmt.Sprintf("%v %v:00", submitRequest.Submission[DateValue], submitRequest.Submission[TimeValue])
+
+		dateValidationError := p.validateDate(fmt.Sprintf("%v", submitRequest.Submission[DateValue]))
+
+		response.Errors = map[string]string{}
+
+		if dateValidationError != "" {
+			response.Errors[DateValue] = dateValidationError
+		}
+
+		timeValidationError := p.validateTime(fmt.Sprintf("%v", submitRequest.Submission[TimeValue]))
+
+		if timeValidationError != "" {
+			response.Errors[TimeValue] = timeValidationError
+		}
+
+		if dateValidationError != "" || timeValidationError != "" {
+			p.returnSubmitDialogResponse(w, response)
+			return
+		}
+	case DateUIType:
+		selectedOption = fmt.Sprintf("%v", submitRequest.Submission[DateValue])
+
+		dateValidationError := p.validateDate(fmt.Sprintf("%v", submitRequest.Submission[DateValue]))
+
+		if dateValidationError != "" {
+			response.Errors = map[string]string{
+				DateValue: dateValidationError,
+			}
+			p.returnSubmitDialogResponse(w, response)
+			return
+		}
+	case TimeUIType:
+		selectedOption = fmt.Sprintf("%v:00", submitRequest.Submission[TimeValue])
+
+		timeValidationError := p.validateTime(fmt.Sprintf("%v", submitRequest.Submission[TimeValue]))
+
+		if timeValidationError != "" {
+			response.Errors = map[string]string{
+				TimeValue: timeValidationError,
+			}
+			p.returnSubmitDialogResponse(w, response)
+			return
+		}
+	}
+
+	client := p.MakeClient(r.Context(), token)
+	if err := client.SendMessageToVirtualAgentAPI(userID, selectedOption, true, &MessageAttachment{}); err != nil {
+		p.API.LogError("Error sending message to VA.", "Error", err.Error())
+		p.returnSubmitDialogResponse(w, response)
+		return
+	}
+
+	newAttachment := []*model.SlackAttachment{}
+	newAttachment = append(newAttachment, &model.SlackAttachment{
+		Text:  fmt.Sprintf("You selected %s: %s", inputType, selectedOption),
+		Color: updatedPostBorderColor,
+	})
+
+	newPost := &model.Post{
+		Id:        postID,
+		ChannelId: submitRequest.ChannelId,
+		UserId:    p.botUserID,
+	}
+
+	model.ParseSlackAttachment(newPost, newAttachment)
+
+	if _, appErr := p.API.UpdatePost(newPost); appErr != nil {
```

Comment:
Instead of updating post, why are we not returning a post? I think that updates the post automatically, right?

Reply:
There are two functions "handleDateTimeSelectionDialog" and "handleDateTimeSelection" if we want to update the post by a return statement it should be in "handleDateTimeSelectionDialog" because this is the function that was invoked when we clicked the button. But we can't return it from this function because till now we don't have the value of the selected date.